map/autoware_lanelet2_map_validator/src/common/validation.cpp

```diff
@@ -76,9 +95,326 @@ std::vector<lanelet::validation::DetectedIssues> validateMap(
+  }
+
+  if (issues.size() > 0) {
+    detected_issues.push_back({target_validator_name, issues});
+  }
+
+  return detected_issues;
+}
+
+void summarize_validator_results(json & json_data)
+{
+  uint64_t warning_count = 0;
+  uint64_t error_count = 0;
+
+  for (auto & requirement : json_data["requirements"]) {
+    std::string id = requirement["id"];
+    bool is_requirement_passed = true;
+    std::map<std::string, bool> validator_results;
+
+    for (const auto & validator : requirement["validators"]) {
+      bool validator_passed = validator["passed"];
+      validator_results[validator["name"]] = validator_passed;
+      is_requirement_passed &= validator_passed;
+
+      if (!validator.contains("issues")) {
+        continue;
+      }
+      for (const auto & issue : validator["issues"]) {
+        if (
+          issue["severity"] ==
+          lanelet::validation::toString(lanelet::validation::Severity::Warning)) {
+          warning_count++;
+        } else if (
+          issue["severity"] ==
+          lanelet::validation::toString(lanelet::validation::Severity::Error)) {
+          error_count++;
+        }
+      }
+    }
+
+    std::cout << BOLD_ONLY << "[" << id << "] ";
+
+    if (is_requirement_passed) {
+      requirement["passed"] = true;
+      std::cout << BOLD_GREEN << "Passed" << FONT_RESET << std::endl;
+    } else {
+      requirement["passed"] = false;
+      std::cout << BOLD_RED << "Failed" << FONT_RESET << std::endl;
+    }
+
+    for (const auto & [name, result] : validator_results) {
+      if (result) {
+        std::cout << "  - " << name << ": " << NORMAL_GREEN << "Passed" << FONT_RESET << std::endl;
+      } else {
+        std::cout << "  - " << name << ": " << NORMAL_RED << "Failed" << FONT_RESET << std::endl;
+      }
+    }
+  }
+
+  if (warning_count + error_count == 0) {
+    std::cout << BOLD_GREEN << "No errors nor warnings were found" << FONT_RESET << std::endl;
+  } else {
+    if (warning_count > 0) {
+      std::cout << BOLD_YELLOW << "Total of " << warning_count << " warnings were found"
+                << FONT_RESET << std::endl;
+    }
+    if (error_count > 0) {
+      std::cout << BOLD_RED << "Total of " << error_count << " errors were found" << FONT_RESET
+                << std::endl;
+    }
+  }
+}
+
+lanelet::validation::ValidationConfig replace_validator(
+  const lanelet::validation::ValidationConfig & input, const std::string & validator_name)
+{
+  auto temp = input;
+  temp.checksFilter = validator_name;
+  return temp;
+}
+
+void process_requirements(json json_data, const MetaConfig & validator_config)
+{
+  std::vector<lanelet::validation::DetectedIssues> issues;
+
+  // List up validators in order
+  Validators validators = parse_validators(json_data);
+  auto [validation_queue, remaining_validators] = create_validation_queue(validators);
+
+  // Note validators that cannot be run from the start
+  std::vector<lanelet::validation::DetectedIssues> unused_validator_issues =
```

Comment:
I recommend because moved objects should not remain in the scope as it is left in a valid but not-guaranteed state.

```suggestion
  if (auto unused_validator_issues = descript_unused_validators_to_json(json_data, remaining_validators); !unused_validator_issues.empty()) {
    appendIssues(...)
  }
```

Reply:
Thanks for your comments!
Fixed in here https://github.com/autowarefoundation/autoware_tools/pull/147/commits/be45043c11907a331c5df05d3926a766f445bfbb